### PR TITLE
[distributed] using code from eventing, that got moved there

### LINKED ISSUE #398: Commonly used checkRetry override
**Problem**

The distributed channel has a custom override of the `checkRetry`:
https://github.com/knative-sandbox/eventing-kafka/blob/c41cdae9688ffabf024e575be772096369c3cd3d/pkg/channel/distributed/dispatcher/dispatcher/handler.go#L157-L200

The method is very detailed and covers a lot of cases, than vendored the "default" implementation:

https://github.com/knative-sandbox/eventing-kafka/blob/c41cdae9688ffabf024e575be772096369c3cd3d/vendor/knative.dev/eventing/pkg/kncloudevents/message_sender.go#L163

Question: Should we move this up? to the `knative.dev/eventing/pkg/kncloudevents` package? Or also use this on the "consolidated" channel?

**[Persona:](https://github.com/knative/eventing/blob/master/docs/personas.md)**
Which persona is this feature for?

**Exit Criteria**
A measurable (binary) test that would indicate that the problem has been resolved.

**Time Estimate (optional):**
How many developer-days do you think this may take to resolve?

**Additional context (optional)**
Add any other context about the feature request here.

### PR DESCRIPTION
**Pull request by matzew**
Signed-off-by: Matthias Wessendorf <mwessend@redhat.com>

Fixes #398

## Proposed Changes

- refacting, leveraging the function which has been moved 
-
-

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

File: pkg/channel/distributed/dispatcher/dispatcher/handler.go [hunk 1]
```diff
@@ -147,54 +146,3 @@ func (h *Handler) consumeMessage(context context.Context, consumerMessage *saram
...
-		logger.Warn("Failed To Send Message To Subscriber Service - Not Retrying")
-		return false, nil
-	} else if statusCode == -1 {
-		logger.Warn("No StatusCode Detected In Error - Retrying")
-		return true, nil
-	}
-
-	// Do Not Retry 1XX, 2XX, & Most 4XX StatusCode Responses
-	return false, nil
-}
```
Comment (travis-minke-sap) [inline pkg/channel/distributed/dispatcher/dispatcher/handler.go:200]: So... what's the rationale for moving this function into eventing ?   I had always viewed this as the distributed KafkaChannel's opinionated view of "when" to retry.  The RetryConfig exposes this hook for implementations to customize the retry decision.

Are you planning on applying the same logic to the consolidated KafkaChannel or using it somewhere else?  Is this part of the larger plan to expose configuration in the Delivery block for the retry decision?  I'm ok with it living in eventing, as long as we can make changes to it in the interim before we have this full configuration capability.

Comment (matzew) [reply pkg/channel/distributed/dispatcher/dispatcher/handler.go:200]: > Are you planning on applying the same logic to the consolidated KafkaChannel

yes, that's the plan

Comment (matzew) [reply pkg/channel/distributed/dispatcher/dispatcher/handler.go:200]: and yes - we can make changes - sure

Comment (matzew) [reply pkg/channel/distributed/dispatcher/dispatcher/handler.go:200]: I like this implementation much better - and would even want to deprecate the old one, in the long run, hence the motivation for moving out

Comment (travis-minke-sap) [reply pkg/channel/distributed/dispatcher/dispatcher/handler.go:200]: Ok sounds good - I just saw same discussion in issue #398 - sorry for duplicating here - I was out for a couple days and wasn't paying attention (obviously : )

**Pull request merged**